### PR TITLE
build custom docker image; run es in docker-compose

### PR DESCRIPTION
**Pull request by shubham-een**
- Added docker directory for custom docker build
- Run ES cluster in docker-compose

### CODE REVIEW

File: elasticsearch/docker/docker-compose-1node.yaml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+services:
+  setup:
+    # image: docker.elastic.co/elasticsearch/elasticsearch:${STACK_VERSION}
+    image: ${DOCKER_IMAGE}
+    volumes:
+      - certs:/usr/share/elasticsearch/config/certs
+    user: "0"
+    command: >
+      bash -c '
```
Comment (ashwinikmp) [inline elasticsearch/docker/docker-compose-1node.yaml:13]: Do we need certs and etc for single node cluster ?

File: elasticsearch/docker/docker-compose-3nodes.yaml [hunk 2]
```diff
@@ -0,0 +1,267 @@
+## Ref: https://www.elastic.co/guide/en/elastic-stack-get-started/current/get-started-stack-docker.html
```
Comment (ashwinikmp) [inline elasticsearch/docker/docker-compose-3nodes.yaml:1]: Do we need 3 node docker cluster for regular use ? Are we going to maintain it ?

File: elasticsearch/docker/docker-compose-1node.yaml (hunk 1 above)
Comment (ashwinikmp) [reply elasticsearch/docker/docker-compose-1node.yaml:13]: See if this alone is sufficient ?

```
version: '3.4'

x-logging: 
  &default-logging
  driver: "json-file"
  options:
    max-size: 20m
    max-file: "5"

services:

  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.1.0
    restart: always
    logging: *default-logging
    environment:
      ES_JAVA_OPTS: "-Xmx256m -Xms256m"
      discovery.type: single-node
      xpack.monitoring.collection.enabled: true
      xpack.security.enabled: false
    ports:
      - "9200:9200"
      - "9300:9300"
    volumes:
      - es_data:/data
    deploy:
      mode: replicated
      replicas: 1

  kibana:
    image: docker.elastic.co/kibana/kibana:8.1.0
    restart: always
    logging: *default-logging
    ports:
      - "5601:5601"
    environment:
      ELASTICSEARCH_URL: http://elasticsearch:9200
      ELASTICSEARCH_HOSTS: '["http://elasticsearch:9200"]'
    deploy:
      mode: replicated
      replicas: 1


volumes:
    es_data:
```

**Review by ashwinikmp — COMMENTED** [review]
Please check comments

File: elasticsearch/docker/docker-compose-3nodes.yaml (hunk 2 above)
Comment (shubham-een) [reply elasticsearch/docker/docker-compose-3nodes.yaml:1]: yes, we can remove that.

File: elasticsearch/docker/docker-compose-1node.yaml (hunk 1 above)
Comment (shubham-een) [reply elasticsearch/docker/docker-compose-1node.yaml:13]: yes this is simple and works, s3 plugin is now included in elasticsearch docker so we needn't build a custom image.

**Comment by ashwinikmp** [comment]
@shubham-een please let me know if the PR is ready for review. I see that you have removed dockerfile. I think that could be useful in future.

**Comment by ashwinikmp** [comment]
@shubham-een please let me know if the PR is ready for review. I see that you have removed dockerfile. I think that could be useful in future.

**Comment by shubham-een** [comment]
As the plugin was already part of  es docker docker custom build was not required, so removed the dockerfile.
Will add that back.

**Review by ashwinikmp — COMMENTED** [review]
Do we need both docker-compose-1node.yaml and docker-compose.yaml ?

**Comment by shubham-een** [comment]
Removed docker-compose-1node.yaml

**Pull request merged**